js/repeater.js

```diff
@@ -370,8 +455,8 @@
 				});
 				footer.hide();
 			} else {
-				cont = this.infiniteScrollingCont;
-				data = cont.data();
+				var cont = this.infiniteScrollingCont;
+				var data = cont.data();
```

Comment:
Did you leave out the word "beginning"?

Live Time refers not to the variables life in the guts of the computer executing the code, but, rather to the variable's life in the brain of the programmer. Consider the following scenarios:

A) Programmer declares `var foo;` at the top of a long function. Sets `foo` to something near the middle of the function, uses it. Later comes back and sets `foo` to something else near the end of the function, forgetting that it was also used in the middle of the function. Bug.

B) Programmer declares and sets `foo` to something in the middle of the function. Later comes back and tries to declare and set `foo` near the end of the function, get's an eslint warning about `foo` already being declared. Names second `foo` something else. No bug.

Because of this, I believe variables should be declared as close to their first use as is practical/possible. Typically they should be declared at the moment they are set. 

And yes I understand hoisting blah blah blah. I have found and fixed several bugs as a result of people being terrified of hoisting or trying to be clever and showing they know about hoisting and declaring all their variables at the top of a function. It's dangerous and unnecessary. If the compiler does it anyways, we certainly don't need to. Especially when ESLint is there to catch it if you try and re-declare a variable.

$0.02 😁